atef/check.py

```diff
@@ -0,0 +1,707 @@
+from __future__ import annotations
+
+import enum
+import logging
+from dataclasses import dataclass, field
+from typing import (Any, Callable, ClassVar, Dict, Generator, List, Optional,
+                    Sequence, Tuple, Type, Union)
+
+import numpy as np
+import ophyd
+
+from . import serialization
+
+Number = Union[int, float]
```

Comment:
I'm surprised this isn't a built-in, it would get a lot of use...

Reply:
There is `numbers.Number`, but for whatever reason my static type checking tool was raising errors on valid inputs. It may have been on my end, and I'd be curious to hear what you see in your setup.

Reply:
I think I was also having trouble with that one last time I tried (in that big daq pr)